lldp_util.c

```diff
@@ -1055,9 +1055,11 @@ int get_saddr6(const char *ifname, struct sockaddr_in6 *saddr)
 
 	rc = getifaddrs(&ifaddr);
 	if (rc == 0) {
+		rc = -1;
 		for (ifa = ifaddr; ifa != NULL; ifa = ifa->ifa_next) {
-			if ((ifa->ifa_addr->sa_family == AF_INET6) &&
-			    (strncmp(ifa->ifa_name, ifname, IFNAMSIZ) == 0)) {
```

Comment:
A strncmp() should always compared against zero as it returns a tristate value and not a boolean. It is much harder to determine the author's intent with that comparison is omitted.

Reply:
In this case we only care that the interface is exactly the same name, so while strncmp will tell us which side of the string was 'greater' it isn't relevant for the code in question.  Indeed, a common idiom is:
  if (!strncmp(...))
rather than an explicit test for 0, since the 'not' is readable, conserves horizontal space, and conveys the same information.  If it mattered which side of the string were greater (for example because the code were making decisions based on a tree structure) then an explicit check for >, <, and = would make sense.  Here it does not.